fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ReduceAggregateChildOutputRows.java

```diff
@@ -0,0 +1,88 @@
+import org.apache.doris.nereids.trees.expressions.functions.agg.Max;
+import org.apache.doris.nereids.trees.expressions.functions.agg.Min;
+import org.apache.doris.nereids.trees.plans.LimitPhase;
+import org.apache.doris.nereids.trees.plans.logical.LogicalAggregate;
+import org.apache.doris.nereids.trees.plans.logical.LogicalLimit;
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** ReduceAggregateChildOutputRows
+ * with group by:
+ * select max(1) from t1 group by c1; -> select 1 from (select c1 from t1 group by c1);
+ * without group by:
+ * select max(1) from t1; -> select max(1) from (select 1 from t1 limit 1) tmp;
+ * */
+public class ReduceAggregateChildOutputRows extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalAggregate().then(agg -> {
+            Set<AggregateFunction> aggFunctions = agg.getAggregateFunctions();
+            // check whether we have aggregate(constant) in all aggregateFunctions
+            if (!agg.isRewrittenByReduceAggregateChildOutputRows()
+                    || aggFunctions.isEmpty() || !aggFunctions.stream().allMatch(
+                        f -> (f instanceof Min || f instanceof Max)
+                            && (f.arity() == 1 && f.child(0).isConstant()))) {
+                return null;
+            }
+
+            ImmutableList.Builder<NamedExpression> newOutput = ImmutableList.builder();
+            for (int i = 0; i < agg.getOutputExpressions().size(); i++) {
+                NamedExpression expr = agg.getOutputExpressions().get(i);
+                if (expr instanceof Alias && expr.child(0) instanceof AggregateFunction) {
+                    AggregateFunction f = (AggregateFunction) expr.child(0);
+                    if (f instanceof Min || f instanceof Max) {
+                        newOutput.add(new Alias(expr.getExprId(), f.child(0), expr.getName()));
+                    } else {
+                        throw new AnalysisException("Unexpected aggregate function: " + f);
+                    }
+                } else {
+                    newOutput.add(expr);
+                }
+            }
+
+            if (agg.getGroupByExpressions().isEmpty()) {
+                LogicalAggregate newAgg = new LogicalAggregate<>(agg.getGroupByExpressions(),
+                        agg.getOutputExpressions(), new LogicalLimit(1, 0, LimitPhase.ORIGIN,
+                                new LogicalProject<>(newOutput.build(), agg.child())));
+                newAgg.setRewrittenByReduceAggregateChildOutputRows(true);
+                return newAgg;
+            } else {
+                List<NamedExpression> childOutput = agg.getGroupByExpressions().stream().map(expr ->
+                        (NamedExpression) expr).collect(Collectors.toList());
```

Comment:
avoid stream api, use ImmutableList. and maybe use remove generic trick could avoid generate new list here?